srtcore/core.cpp

```diff
@@ -10296,8 +10410,21 @@ SRT_REJECT_REASON CUDT::processConnectRequest(const sockaddr_any& addr, CPacket&
             HLOGC(mglog.Debug,
                   log << CONID() << "processConnectRequest: sending ABNORMAL handshake info req="
                       << RequestTypeStr(hs.m_iReqType));
+            bool has_message =! streaminfo_msg.empty();
+
+            if (has_message)
+            {
+                hs.m_iType |= CHandShake::HS_EXT_CONFIG;
+            }
+
             size_t size = CHandShake::m_iContentSize;
             hs.store_to((packet.m_pcData), (size));
+
+            if (has_message)
+            {
+                size = addHandshakeExtension((packet.m_pcData), SRT_CMD_SID, size, streaminfo_msg);
+            }
+            packet.setLength(size);
```

Comment:
There is a bugfix here. This length wasn't set anywhere here, it was set to the plain handshake size + 4 bytes (unknown source), hence it passed the test to not be <= the size. This time it is set to the real size of the handshake, with extension if needed.

Reply:
#1273 reports this problem.